contracts/Vote.sol

```diff
@@ -333,6 +343,48 @@ contract Vote is UUPSOwnableUpgradeable, UsingRegistryUpgradeable, Managed {
         return stCelo;
     }
 
+    /**
+     * Deletes proposalId from voter's history if proposal expired.
+     * @param voter The voter address.
+     * @param proposalId The proposal id.
+     * @param index Index of voter's proposal id.
+     */
+    function deleteExpiredVoterProposalId(
+        address voter,
+        uint256 proposalId,
+        uint256 index
+    ) external {
+        Voter storage voterStruct = voters[voter];
+
+        uint256 proposalIdOnChain = voterStruct.votedProposalIds[index];
+        if (proposalIdOnChain == 0 || proposalIdOnChain != proposalId) {
+            revert IncorrectIndex();
+        }
+
+        deleteExpiredProposalTimestamp(proposalId);
```

Comment:
This should be executed when `proposalTimestamp != 0`, otherwise, it will always try to delete the timestamp, even if already deleted.